dm/dm/worker/status.go

```diff
@@ -123,3 +124,25 @@ func (w *SourceWorker) GetUnitAndSourceStatusJSON(stName string, sourceStatus *b
 	}
 	return s
 }
+
+func (w *SourceWorker) GetValidateStatus(stName string, filterStatus string) []*pb.ValidationStatus {
+	sts := w.subTaskHolder.getAllSubTasks()
+	res := make([]*pb.ValidationStatus, 0)
+	if len(sts) == 0 {
+		return res
+	}
+	sourceIP := w.cfg.DecryptPassword().From.Host
+	for _, st := range sts {
+		if st.cfg.Name != stName || st.validator == nil || !st.validator.Started() {
+			continue
+		}
+		tblStats := st.validator.GetValidationStatus()
+		for _, stat := range tblStats {
+			if filterStatus == "" || strings.EqualFold(stat.ValidationStatus, filterStatus) {
```

Comment:
use `==` directly?

Reply:
`filterStatus` is possibly of lower case. `EqualFold` will compare the two strings case insensitively.